### ISSUE TITLE
Add support for serialization of object members (using prefixing of parent name)

### ISSUE THREAD

**Issue by fenjen** [FasterXML/jackson-dataformat-csv#9]
Let's assume we have two classes:

``` java
class A {
  String s1 = "A_string1";
  String s2 = "A_string2";
  @JsonIgnore B b;
}
```

``` java
class B {
  String s1 = "B_string1";
  String s2 = "B_string2";
}
```

If we'd serialize an instance of `A`  the result is `A_string1,A_string2`
The serialization of an instance of B would look like this `B_string1,B_string2`

Now If we'd remove the `@JsonIgnore` from class A, we'd get the Exception:
`com.fasterxml.jackson.core.JsonGenerationException: CSV generator does not support Object values for properties`

Why couldn't we just serialize b in a -- the result would be quite intuitive: `A_string1,A_string2,B_string1,B_string2`

This would enable jackson-csv to serialize object members as it's possible in XML and JSON.

**Comment by kringol** [comment]
Is there any pseudo-easy way to achieve this or similar nowadays by using a list of JsonPointer paths for colums and let the csvmapper write those nodes to the csv, provided that the resulting node of the pointer is of a serializable type and not an object. 
JsonNode already supports doing node.at("/rootNode/objectField/name" or even with array positions

**Comment by cowtowncoder** [comment]
No. Jackson stays out of transformation business as a boundary. You can achieve this outside of core databind by creating a JsonNode, transforming it, but there is nothing automated and I have no plans to add anything that requires building of intermediate tree structures.

But as I said `@JsonUnwrapped` does work for flattening, for cases where that is applicable.

**Comment by zadeswapnilzade** [comment]
this is going to be in next release ?

**Comment by cowtowncoder** [comment]
@zadeswapnilzade There is no active development for this feature at this point in time.

**Comment by don-han** [comment]
@JsonUnwrapped doesn't work if we are using the same POJO for deserializing from JSON and serializing to CSV [as @5kilodope pointed out](https://github.com/FasterXML/jackson-dataformat-csv/issues/9#issuecomment-32590395). 

It seems like [mix-ins](http://demeranville.com/writing-csv-using-jackson-csvmapper-mixin-annotations/) might be able to solve this problem, but given that the conversion between JSON and CSV are probably the main use for this library, I think this would be a very valuable feature.

**Comment by cowtowncoder** [comment]
@don-han Certainly would be valuable, if someone had time to work on this.

**Comment by jmatthewpryor** [comment]
One simple, but manual solution might be to allow dot notation in the Schema Builder

```
CsvSchema schema = CsvSchema.builder()
        .addColumn("person.firstName")
        .addColumn("person.firstName")
        .addColumn("person.age", CsvSchema.ColumnType.NUMBER)
        .build();
```

That would allow for writing CSV from complex objects

**Comment by cowtowncoder** [comment]
@jmatthewpryor there are couple of related challenges: but perhaps something like this could work, similar to how `jackson-dataformat-properties` does things (by keeping track of prefix by object nesting).

**Comment by alianos-** [comment]
@don-han mixins seems deprecated.

**Comment by cowtowncoder** [comment]
@alianos- Mix-ins are certainly not deprecated as a feature. But maybe you mean something else?

**Comment by alianos-** [comment]
If I write `mapper.addMixInAnnotations(Person.class, PersonFormat.class);` my IDE marks the method as deprecated. Tbh I didn't look more into it since I didn't think it can help with the issue at hand.

**Comment by ptahchiev** [comment]
+1 for this :)

**Comment by ptahchiev** [comment]
Guys, I have created a pull-request for this here: https://github.com/FasterXML/jackson-dataformats-text/pull/97 however I have one test failing. I feel like I'm almost there, perhaps @cowtowncoder can help me out :)

**Comment by ptahchiev** [comment]
I have updated the PR and now all the tests are passing. However, I am not yet 100% sure what in what I have implemented. @cowtowncoder  can you please have a look at it and if there's something else to be done, please let me know.
